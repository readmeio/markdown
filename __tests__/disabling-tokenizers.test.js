const markdown = require('../index');

describe('disableTokenizers: "inlines"', () => {
  const opts = { disableTokenizers: 'inlines' };

  it('disabling inlineCode', () => {
    const md = '`const js = true `';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });

  it('disabling emphasis', () => {
    const md = '*emphatic **strong** emphatic*';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });

  it('disabling delete', () => {
    const md = '~~strikethrough~~';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });
});

describe('disableTokenizers: "blocks"', () => {
  const opts = { disableTokenizers: 'blocks' };

  it('disabling block tokenizer', () => {
    const md = '# heading 1';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });
});

describe('disableTokenizers: {}', () => {
  it('disables a block tokenizer', () => {
    const opts = { disableTokenizers: { block: ['atxHeading'] } };
    const md = '# heading 1';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });

  it('disables an inline tokenizer', () => {
    const opts = { disableTokenizers: { inline: ['code'] } };
    const md = '`const js = true`';
    expect(markdown.mdast(md, opts)).toMatchSnapshot();
  });
});
