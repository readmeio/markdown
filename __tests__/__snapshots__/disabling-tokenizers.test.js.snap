// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disableTokenizers: "blocks" disabling block tokenizer 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "#  heading 1",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`disableTokenizers: "inlines" disabling delete 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "~~strikethrough~~",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`disableTokenizers: "inlines" disabling emphasis 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "*emphatic **strong** emphatic*",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`disableTokenizers: "inlines" disabling inlineCode 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "\`const js = true \`",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`disableTokenizers: {} disables a block tokenizer 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "#  heading 1",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`disableTokenizers: {} disables an inline tokenizer 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "type": "text",
          "value": "\`const js = true\`",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
